app/controllers/candidate_interface/decisions_controller.rb

```diff
@@ -72,5 +67,21 @@ def check_that_candidate_can_decline
         render_404
       end
     end
+
+    def check_that_candidate_can_accept
+      unless ApplicationStateChange.new(@application_choice).can_accept?
+        render_404
+      end
+    end
+
+    def check_that_candidate_can_withdraw
+      unless ApplicationStateChange.new(@application_choice).can_withdraw?
+        redirect_to candidate_interface_application_form_path
```

Comment:
I think 404 is more appropriate here and in `check_that_candidate_has_an_offer`, is that okay?

Reply:
Yep, I'm just about to change all of them 404 as we discussed on the previous PR 👍